core/src/main/scala/org/apache/spark/sql/rapids/tool/AppBase.scala

```diff
@@ -69,14 +69,18 @@ abstract class AppBase(
 
   var gpuMode = false
 
+  val isMLOPsEnabled = ToolUtils.isMLOpsEnabled
+
   def getOrCreateStage(info: StageInfo): StageInfoClass = {
     val stage = stageIdToInfo.getOrElseUpdate((info.stageId, info.attemptNumber()),
       new StageInfoClass(info))
 
     // check if there are any SparkML/XGBoost functions or expressions
-    val mlOps = checkMLOps(stage.info.details)
-    if (mlOps.nonEmpty) {
-      stage.mlOps = Some(mlOps)
+    if (isMLOPsEnabled) {
```

Comment:
we should not be looking for mlops here, as the function name implies this can be called multiple times so we could be parsing it multiple times when we don't need to.  This should move out to be called after the fact.  All you are doing here is adding in extra data that takes up memory, process this before we create the qualification tool output. Hopefully moving it there makes passing the config to turn on/off easier

Reply:
for example in QualificaitonAppInfo.aggregateStats you can access like:
     ```
 stageIdToInfo.map { case (_, stageInfo) =>
        stageInfo.info.details
      }
```

Reply:
Thanks for the suggestion. Moved the parsing code to QualificationSummaryInfo. PTAL.